src/Fixer/FixerInterface.php

```diff
@@ -54,6 +54,12 @@ public function fix(\SplFileInfo $file, Tokens $tokens): void;
      */
     public function getDefinition(): FixerDefinitionInterface;
 
+    /**
+     * Returns the absolute URI [RFC3986](http://www.rfc-editor.org/info/rfc3986)
+     * of the primary documentation for the fixer.
+     */
+    public function getHelpUri(): string;
```

Comment:
Since `FixerInterface` is not internal (it's a contract for custom fixers too) this is a breaking change. Is it required to get this info straight from the fixers, or could it be done using an external class?

Reply:
I've recently released https://github.com/llaville/sarif-php-sdk/releases/tag/1.2.0 that will add SARIF support to most common PHP linters.

I've prepared a `\Bartlett\Sarif\Converter\PhpCsFixerConverter` for PHP-CS-Fixer too, that passed all validation by official [Sarif Validator](https://sarifweb.azurewebsites.net/Validation)
But to be able to use it, the https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/discussions/7995 should be accepted ;-)

All, to said that SARIF used an [helpUri](https://docs.oasis-open.org/sarif/sarif/v2.1.0/errata01/os/sarif-v2.1.0-errata01-os-complete.html#_Toc141791098) property to enforce online documentation.

This is also my goal with PHP-CS-Fixer.

Reply:
I understand that, but it does not change the fact it's a breaking change and requires major release, which we don't want now, as it would require massive amount of work to clean the codebase from deprecations ☺️. And it would require @keradus' review and approval, which often slows things down because of his limited availability. With non-invasive approach I think we can continue without him.

Having said that, we could make it part of a contract, but using composition - we can introduce a new interface, dedicated only for that, implement it internally to satisfy the reporter's needs, but at the same time let people implement it at their own pace. That would require checking if the fixer implements this interface while preparing the output and linking only these rules that provide a doc URL.

Reply:
Agree with you to do it, but I'm sorry I won't have enough free time today !

Reply:
Not a problem, everyone' busy. Let's continue as fast as possible, but without any rush ☺️.

Reply:
@Wirone I'll work now on your idea to split interfaces to have something like : 

```php
<?php
namespace PhpCsFixer\Fixer;

interface FixerReportInterface
{
    public function getHelpUri(): string;

    public function getExtraInfoFixers(): array;
}
```
The `getExtraInfoFixers` will return any additional info as `helpUri` (e.g), but much more.

This interface will be usefull, in state only for `TextReporter`, because other Reporter don't provide online doc by format spec !